vlm/src/main/scala/geotrellis/contrib/vlm/RasterSummary.scala

```diff
@@ -0,0 +1,100 @@
+package geotrellis.contrib.vlm
+
+import geotrellis.proj4.CRS
+import geotrellis.raster.{CellGrid, CellSize, CellType, RasterExtent}
+import geotrellis.spark._
+import geotrellis.spark.tiling.{LayoutDefinition, LayoutLevel, LayoutScheme}
+import geotrellis.util._
+import geotrellis.vector.{Extent, ProjectedExtent}
+import org.apache.spark.rdd.RDD
+
+case class RasterSummary(
+  crs: CRS,
+  cellType: CellType,
+  cellSize: CellSize,
+  extent: Extent,
+  cells: Long,
+  count: Long
+) {
+
+  def estimatePartitionsNumber: Int = {
+    import squants.information._
+    val bytes = Bytes(cellType.bytes * cells)
+    val numPartitions: Int = math.max((bytes / Megabytes(64)).toInt, 1)
+    println(s"Using $numPartitions partitions for ${bytes.toString(Gigabytes)}")
+    numPartitions
+  }
+
+  def levelFor(layoutScheme: LayoutScheme): LayoutLevel =
+    layoutScheme.levelFor(extent, cellSize)
+
+  def toRasterExtent: RasterExtent = RasterExtent(extent, cellSize)
+
+  def layoutDefinition(scheme: LayoutScheme): LayoutDefinition = scheme.levelFor(extent, cellSize).layout
+
+  def combine(other: RasterSummary): RasterSummary = {
+    require(other.crs == crs, s"Can't combine LayerExtent for different CRS: $crs, ${other.crs}")
+    val smallestCellSize = if (cellSize.resolution < other.cellSize.resolution) cellSize else other.cellSize
+    RasterSummary(
+      crs,
+      cellType.union(other.cellType),
+      smallestCellSize,
+      extent.combine(other.extent),
+      cells + other.cells,
+      count + other.count
+    )
+  }
+
+  def toTileLayerMetadata(layoutLevel: LayoutLevel): (TileLayerMetadata[SpatialKey], Some[Int]) = {
+    val LayoutLevel(zoom, layoutDefinition) = layoutLevel
+    val re = TargetGrid(layoutDefinition)(toRasterExtent)
+    val dataBounds = KeyBounds(layoutDefinition.mapTransform.extentToBounds(re.extent))
+    TileLayerMetadata[SpatialKey](cellType, layoutDefinition, re.extent, crs, dataBounds) -> Some(zoom)
+  }
+
+  def toTileLayerMetadata(layoutDefinition: LayoutDefinition, zoom: Int): (TileLayerMetadata[SpatialKey], Some[Int]) =
+    toTileLayerMetadata(LayoutLevel(zoom, layoutDefinition))
+
+  def toTileLayerMetadata(layoutType: LayoutType): (TileLayerMetadata[SpatialKey], Option[Int]) = {
+    val (ld, zoom) = layoutType.layoutDefinitionWithZoom(crs, extent, cellSize)
+    val dataBounds: Bounds[SpatialKey] = KeyBounds(ld.mapTransform.extentToBounds(extent))
+    TileLayerMetadata[SpatialKey](cellType, ld, extent, crs, dataBounds) -> zoom
+  }
+
+  def resample(resampleGrid: ResampleGrid): RasterSummary = {
```

Comment:
Should this be moved to `RasterSummarySpec`? Since earlier we talked about how it doesn't make much sense for `RasterSummary` to have a `resample` method.

Reply:
I think this function should still be there, as it can be very useful for the test purposes, at least for now.

Reply:
That makes sense. I think we can move it to to the tests at some point instead of removing it.

Reply:
@jbouffard that's true I added a TODO thing, probably i will remove it in a separate PR!